python/dgl/sparse/sparse_matrix.py

```diff
@@ -565,6 +568,10 @@ def from_csr(
     if val is None:
         val = torch.ones(indices.shape[0]).to(indptr.device)
 
+    assert (
```

Comment:
Move assert above the first if check.

Reply:
This check should be after `if val is None`.